### PR TITLE
Refund deposits after curation window (#63)

### LINKED ISSUE #63: Return deposits after curation window

### PR DESCRIPTION
**Pull request by shanev**
Fixes #63 
Depends on #53 

After each  block:
* Iterate posts with expired curation windows
* Refund post creator
* Iterate curator for each post and refund each

Also hashing post id.

### CODE REVIEW

File: x/curating/keeper/post.go [hunk 1]
```diff
@@ -142,3 +153,39 @@ func hash(body string) ([]byte, error) {
...
+	ctx sdk.Context, cb func(post types.Post) (stop bool)) {
+
+	it := k.CurationQueueIterator(ctx, ctx.BlockTime())
+	defer it.Close()
+
+	for ; it.Valid(); it.Next() {
+		vps := types.VPPairs{}
+		k.cdc.MustUnmarshalBinaryBare(it.Value(), &vps)
+		for _, vp := range vps.Pairs {
+			post, found, err := k.GetPostZ(ctx, vp.VendorID, vp.PostID)
```
Comment (jhernandezb) [inline x/curating/keeper/post.go:169]: is `vp.PostID` the hash ?

File: x/curating/keeper/upvote.go [hunk 2]
```diff
@@ -44,14 +44,14 @@ func (k Keeper) CreateUpvote(
 		if err != nil {
 			return err
 		}
+		// shadow deposit as its no longer available
+		deposit = deposit.Sub(deposit)
```
Comment (jhernandezb) [inline x/curating/keeper/upvote.go:48]: adding a comment to this part (for some reason can't comment)
`err = k.CreatePost(ctx, vendorID, postID, "", deposit, curator, rewardAccount)` 
Now that I think it , shouldn't the post already exists ? I mean if the post doesn't previously exists on chain a transaction should be submitted before from the creator with `reward_address = curator` 

The conflict is that the post would not contain the real `author` address

File: x/curating/keeper/post.go (hunk 1 above)
Comment (shanev) [reply x/curating/keeper/post.go:169]: I was being lazy and you caught it lol

File: x/curating/keeper/upvote.go (hunk 2 above)
Comment (shanev) [reply x/curating/keeper/upvote.go:48]: Ah yes, the post would not contain the real author/creator in this case. Maybe we should make the creator optional? So in this case, the creator would be `nil`, but the `reward_address` would be the curator.

Comment (shanev) [reply x/curating/keeper/upvote.go:48]: Created #76

**Pull request merged**